### PR TITLE
Recall button

### PR DESCRIPTION
**Pull request by jl-codes**
Webview:
Screen Shot 2019-08-09 at 1 41 18 PM

### CODE REVIEW

File: Recipick/ViewController.swift [hunk 1]
```diff
@@ -8,13 +8,27 @@
 
 import UIKit
```
Comment (prrenfrew) [inline Recipick/ViewController.swift]: No space between imports.

File: Recipick/ViewController.swift [hunk 2]
```diff
@@ -8,13 +8,27 @@
...
+class ViewController: UIViewController {
+  
+  let recallURL = Constants.recallsURL
+  
+  @IBAction func recentRecalls(_ sender: UIButton) {
+    let recalls = recallURL
+    showSafariVC(for: recalls)
+  }
+  
+  func showSafariVC(for url: String) {
```
Comment (prrenfrew) [inline Recipick/ViewController.swift]: Seems like a private function.

File: Recipick/ViewController.swift [hunk 3]
```diff
@@ -8,13 +8,27 @@
 
 import UIKit
 
-class ViewController: UIViewController {
+import SafariServices
 
+class ViewController: UIViewController {
+  
+  let recallURL = Constants.recallsURL
```
Comment (prrenfrew) [inline Recipick/ViewController.swift]: Any reason this is not private?

File: Recipick/ViewController.swift [hunk 4]
```diff
@@ -8,13 +8,27 @@
...
+  
+  @IBAction func recentRecalls(_ sender: UIButton) {
+    let recalls = recallURL
```
Comment (prrenfrew) [inline Recipick/ViewController.swift]: No need for this. Just pass recallURL in directly.

File: Recipick/ViewController.swift (hunk 2 above)
Comment (prrenfrew) [reply Recipick/ViewController.swift]: Also, don't use abbreviations. Spell things out so it is easier to know what the abbreviations mean.

File: Recipick/ViewController.swift (hunk 4 above)
Comment (jl-codes) [reply Recipick/ViewController.swift]: Modified, contents of IBAction now reads:
`showSafariViewController(for: Constants.recallsURL)`

File: Recipick/ViewController.swift (hunk 3 above)
Comment (jl-codes) [reply Recipick/ViewController.swift]: The ViewController class is not private, though the `showSafariViewController` function is now private

File: Recipick/ViewController.swift (hunk 2 above)
Comment (jl-codes) [reply Recipick/ViewController.swift]: Modified, func now reads: `private func showSafariViewController(for url: String)`

File: Recipick/ViewController.swift (hunk 1 above)
Comment (jl-codes) [reply Recipick/ViewController.swift]: Fixed

File: Recipick/Constants.swift [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+//
+
+import Foundation
+import SafariServices
+
+enum Constants {
+    static let recallsURL: String = "https://www.fda.gov/safety/recalls-market-withdrawals-safety-alerts"
+}
+
+final class recallsView {
```
Comment (prrenfrew) [inline Recipick/Constants.swift]: Put this in its own file. Also use UpperCamelCase for class names.

Comment (jl-codes) [reply Recipick/Constants.swift]: Resolved location and class name

File: Recipick/ViewController.swift [hunk 6]
```diff
@@ -9,12 +9,15 @@
 import UIKit
 
 class ViewController: UIViewController {
-
+  
+  @IBAction func recentRecalls(_ sender: UIButton) {
+    RecallsViewController.showSafariViewController(viewController: self)
+  }
```
Comment (prrenfrew) [inline Recipick/ViewController.swift]: This was only for testing. Please remove once done with testing.

Comment (jl-codes) [reply Recipick/ViewController.swift]: Removed

**Pull request merged**